recipes/boost/all/conanfile.py

```diff
@@ -1429,7 +1429,7 @@ def add_libprefix(n):
                     libprefix = "lib"
                 return libprefix + n
 
-            all_detected_libraries = set(tools.collect_libs(self))
+            all_detected_libraries = set(l[:-4] if l.endswith(".dll") else l for l in tools.collect_libs(self))
```

Comment:
I've seen this in the passed, https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/libdisasm/all/conanfile.py#L109
https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/argon2/all/conanfile.py#L98

Not sure what the best fix is 🤔

Reply:
My fix would also handle the first one because tools.collect_libs() removes the last extension (so .lib in .dll.lib would get removed, as in .dll.a). But maybe it'd be better to remove all extensions altogether. I'm not an Python expert so I can't conclude anything.

Reply:
@madebr @SpaceIm This is mingw file extension issue, if you can take a peak... not my area of expertise

Reply:
Went and had a closer look, first one is not good because the extension is added to the libname (which is to my knowledge used by conan generators). Second one is very different and specific to the way this package is made so I wouldn't even go and compare.

My fix is specific to the boost recipe where a comparison is made to detect what's been built and if said build was successful, it doesn't directly interfere with Conan tools during packaging and is thus safe (meaning the libraries are correctly copied and packaged in the current version if early checks don't raise false positives).

Conclusion: this PR is relevant and could be used as a reference for future issues regarding MinGW imo.

Reply:
I think the easiest solution would be to remove the `.dll` infix of all import libraries in the `package` method.
```python
def package(self):
    lib_path = os.path.join(self.package_folder, "lib")
    for fn in os.listdir(lib_path):
        if fn.endswith(".dll.a"):
            tools.rename(os.path.join(lib_path, fn), os.path.join(lib_path, "{}.a".format(fn[:-6])))

Reply:
I think it's better to honor filenames extensions, .dll.a is a standard extension in MinGW.

Reply:
Okay. Let's keep the filenames intact then.

Reply:
Thanks guys for helping ❤️